match2/commons/opponent_display.lua

```diff
@@ -207,4 +207,247 @@ function p._createMatchListOpponent(frame, displayType, name, score)
+			flip = props.flip,
+			overflow = props.overflow,
+			player = opponent.players[1],
+			showFlag = props.showFlag,
+			showLink = props.showLink,
+		})
+	end
+end
+
+OpponentDisplay.propTypes.InlineTeam = {
+	flip = 'boolean?',
+	style = TypeUtil.optional(OpponentDisplay.types.TeamStyle),
+	template = 'string',
+}
+
+--[[
+Displays a team as an inline element.
+]]
+function OpponentDisplay.InlineTeam(props)
+	DisplayUtil.assertPropTypes(props, OpponentDisplay.propTypes.InlineTeam)
+
+	local teamExists = mw.ext.TeamTemplate.teamexists(props.template)
+	if props.style == 'standard' or not props.style then
+		if not props.flip then
+			return teamExists
+				and mw.ext.TeamTemplate.team(props.template)
+				or Template.safeExpand(mw.getCurrentFrame(), 'Team', {props.template})
+		else
+			return teamExists
+				and mw.ext.TeamTemplate.team2(props.template)
+				or Template.safeExpand(mw.getCurrentFrame(), 'Team2', {props.template})
+		end
+	elseif props.style == 'short' then
+		if not props.flip then
+			return teamExists
+				and mw.ext.TeamTemplate.teamshort(props.template)
+				or Template.safeExpand(mw.getCurrentFrame(), 'TeamShort', {props.template})
+		else
+			return teamExists
+				and mw.ext.TeamTemplate.team2short(props.template)
+				or Template.safeExpand(mw.getCurrentFrame(), 'Team2Short', {props.template})
+		end
+	elseif props.style == 'bracket' then
+		if not props.flip then
+			return teamExists
+				and mw.ext.TeamTemplate.teambracket(props.template)
+				or Template.safeExpand(mw.getCurrentFrame(), 'TeamBracket', {props.template})
+		else
+			error('Flipped style=bracket is not supported')
+		end
+	end
+end
+
+OpponentDisplay.propTypes.BlockTeam = {
+	flip = 'boolean?',
+	overflow = TypeUtil.optional(DisplayUtil.types.OverflowModes),
+	showLink = 'boolean?',
+	style = TypeUtil.optional(OpponentDisplay.types.TeamStyle),
+	template = 'string',
+}
+
+--[[
+Displays a team as a block element. The width of the component is determined by
+its layout context, and not of the team name.
+]]
+function OpponentDisplay.BlockTeam(props)
+	DisplayUtil.assertPropTypes(props, OpponentDisplay.propTypes.BlockTeam)
+	local style = props.style or 'standard'
+
+	local raw = mw.ext.TeamTemplate.raw(props.template)
+	if not raw then
+		return html.create('div'):addClass('error')
+			:wikitext('No team template exists for name ' .. props.template)
+	end
+
+	local displayName = style == 'standard' and raw.name
+		or style == 'short' and raw.shortname
+		or style == 'bracket' and raw.bracketname
+
+	local nameNode = html.create('span'):addClass('name')
+		:wikitext(props.showLink
+			and '[[' .. raw.page .. '|' .. displayName .. ']]'
+			or displayName
+		)
+	DisplayUtil.applyOverflowStyles(nameNode, props.overflow or 'ellipsis')
+
+	return html.create('div'):addClass('block-team')
+		:addClass(props.showLink == false and 'block-team-hide-link' or nil)
+		:addClass(props.flip and 'flipped' or nil)
+		:node(mw.ext.TeamTemplate.teamicon(props.template))
```

Comment:
I don't know the details of what changed with the new team icons. May require different handling for legacy and non-legacy icons here.